services/ui_backend_service/data/cache/client/cache_async_client.py

```diff
@@ -15,6 +15,7 @@
 
 class CacheAsyncClient(CacheClient):
     _drain_lock = asyncio.Lock()
+    _restart_requested = False
```

Comment:
as this lives in the asyncClient instance, does anything set it back to false after a successful restart? To my understanding the cacheclient instance does not get reinitialized on a restart?

Reply:
Actually it does get reinitialized on restart. This is the relevant line: https://github.com/Netflix/metaflow-service/blob/dcbd511267a45226d4e29d67f1ae2667e58e1147/services/ui_backend_service/data/cache/store.py#L115

Reply:
so it seems, disregard the comment then ;)